### ISSUE TITLE
jest-dom rule for enabled/disabled

### ISSUE THREAD

**Issue by benmonro** [Belco90/eslint-plugin-testing-library#30]
would be nice to have a rule (auto-fixable) that auto-fixes

`expect(element).toHaveProperty('disabled', true)` 

into

`expect(element).toBeDisabled()`

and similar for `toBeEnabled()`

I realize this is part of jest-dom, but jest-dom is in the testing-library org, so it made sense to me to have the rule in here.  could probably come up w/ a bunch of other rules like this for jest-dom.... thoughts?

**Comment by Belco90** [comment]
I'll comment in the corresponding PR, but seeing this rule and #20 I think we should just implement them specifically for `jest-dom` syntax, specifying it properly in the dom and also prefixing the rules with `jest-dom- *`.

**Comment by thomlom** [comment]
It makes sense to support these rules in the plugin as long as the user knows it has to install `jest-dom` to use the rules. Prefixing the rules with `jest-dom` is a great way to communicate this idea.

There are a lot of rule possibilities to implement then, not just `toBeEnabled` and `toBeDisabled`. You could also have the same rule for `toBeRequired`, `toBeVisible`, etc.

I think it would make sense to have a global rule called `prefer-jest-dom-matchers` that could be configurable depending on the matchers we want to enforce, something like:

```json
"testing-library/prefer-jest-dom-matchers": ["warn", ["toBeEnabled", "toBeDisabled", "toBeRequired"]]
```

By default, this rule would include all Jest matchers implemented in the rule. What do you think?

**Comment by Belco90** [comment]
I'm not sure about having a single rule for all that because:

- it makes more sense to add the rules individually so you can set them up individually as they are checking different things. I don't think you should use the rule option for actually setting up which rules to enable. I prefer to have them individually and add a shareable config for jest-dom rules if necessary
- how we could implement all of them in a generic way? For example `toBeInTheDocument` can be used for replacing `expect(element).toBeDefined()` but also for replacing `expect(queryAllBy('foo')).toHaveLength(1)`. I think with your proposal we would apply the wrong approach, as we would be defining rules by jest-dom matchers, when jest-dom should just be the fix for those rules, defined by patterns we want to suggest/replace. So in this case I would have one rule for suggesting `expect(element).toBeDefined()` to be changed, and a different one for suggesting `expect(queryAllBy('foo')).toHaveLength(1)` to be changed, and then use `toBeInTheDocument` to autofix both.

To sum up my thoughts here: it makes more sense to have individual rules prefixed as `jest-dom-*` for patterns we want to suggest and then autofix them with whatever jest-dom matcher for 1) avoiding setting rules by the rule option itself and 2) avoiding grouping different patterns in the same rule.

**Comment by thomlom** [comment]
Of course, totally forgot we could create shareable configs 🤦‍♂ 
Yeah, that makes sense!

**Comment by Belco90** [comment]
Glad to hear this makes sense to you, so what I would do is implement this one as `jest-dom-*` rule and then when we have few more of these rules implemented, we can add proper shareable config for jest-dom. Does that sounds fine? @thomlom @benmonro

**Comment by Belco90** [comment]
Sure, I guess we can add some to our "backlog", tho we already have this one and #20

**Comment by benmonro** [comment]
All sounds good to me. I will push not changes to my PR later today.
